www/js/control/ControlDataTable.jsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { DataTable } from 'react-native-paper';
+import { angularize } from "../angular-react-helper";
+import { array } from "prop-types";
+
+// Note the camelCase to dash-case conventions when translating to .html files!
+// val with explicit call toString() to resolve bool values not showing
+const ControlDataTable = ({ controlData }) => {
+  console.log("Printing data trying to tabulate", controlData);
+  return (
+    <DataTable>
+      {controlData.map((e) =>
+        <DataTable.Row>
+          <DataTable.Cell>{e.key}</DataTable.Cell>
+          <DataTable.Cell>{e.val.toString()}</DataTable.Cell>
+        </DataTable.Row>
+      )}
+    </DataTable>
+    );
+  };
+ControlDataTable.propTypes = {
+    controlData: array
+  }
+ 
+// need call to angularize to let the React and Angular co-mingle
+  //second argument is "module path" - unsure of exact usage, could require update later?
+angularize(ControlDataTable, 'emission.main.control.dataTable'); 
```

Comment:
In AngularJS, everything is registered under a module.

For a typical AngularJS directive/component, you register it under some module path, and then in the places where you need to use it, you list dependencies by their module path.

For React components that are angularized, I made this a little easier - the module path is added to the component itself and can be accessed via `ExampleComponent.module` after `ExampleComponent` is imported. This way, you only have to write the modulepath in one place, and if you make a typo or something it doesn't really matter.

Reply:
Ok, that makes sense, thank you for explaining! I was just emulating what I'd seen other places in the code, now I understand why it works.